controllers/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -141,6 +141,26 @@ func (r *AWSFederatedAccountAccessReconciler) Reconcile(_ context.Context, reque
 	}
 
 	if currentFAA.Status.State != "" {
+		// Make sure the awsFederatedRoleName label is present
+		if !hasLabel(currentFAA, awsv1alpha1.FederatedRoleNameLabel) {
+			reqLogger.Info(fmt.Sprintf("Adding %s label with value %s to AccountAccess %s", awsv1alpha1.FederatedRoleNameLabel, requestedRole.Name, currentFAA.Name))
+
+			newLabel := map[string]string{awsv1alpha1.FederatedRoleNameLabel: requestedRole.Name}
+
+			// Join the new UID label with any current labels
+			if currentFAA.Labels != nil {
+				currentFAA.Labels = controllerutils.JoinLabelMaps(currentFAA.Labels, newLabel)
+			} else {
+				currentFAA.Labels = newLabel
+			}
```

Comment:
```suggestion
			// Join the new UID label with any current labels
			currentFAA.Labels = newLabel
			if currentFAA.Labels != nil {
				currentFAA.Labels = controllerutils.JoinLabelMaps(currentFAA.Labels, newLabel)
			} 
```

Minor nit, this is more "Go-like" to try and avoid an else

Reply:
This would override the existing map on the CR, however I think we could update `JoinLabelMaps` to explicitly handle null maps. Let me take a look...